connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -37,15 +37,15 @@ import org.apache.spark.connect.proto
  */
 class RelationalGroupedDataset private[sql] (
     private[sql] val df: DataFrame,
-    private[sql] val groupingExprs: Seq[proto.Expression],
+    private[sql] val groupingExprs: Seq[Column],
```

Comment:
What prompted the change to `Column`?

Reply:
To simply the `keysFunc` when creating `KeyValueGroupedDatasetImpl` ([code](https://github.com/apache/spark/pull/40796/files#diff-91fc5eb4a3a5099e53b0e14b27b4cfc427c2df98612844f214a5c778b36d797fR632)):
```
    new KeyValueGroupedDatasetImpl(
      sparkSession,
      plan,
      kEncoder,
      kEncoder,
      vEncoder,
      vEncoder,
      (Seq(dummyGroupingFunc) ++ groupingExprs).map(_.expr).asJava,
      UdfUtils.identical(),
      () => df.select(groupingExprs: _*).as(kEncoder)) // To avoid creating new cols again.
  }
```